src/util/log.cpp

```diff
@@ -100,14 +116,44 @@ static std::string sprintf(const char *msg, va_list arglist) {
 /*
  * Initialize the logging
  */
+void Log::logToFile() {
+	std::lock_guard<std::mutex> guard(log_mutex);
+    Log::LogLevel level = levelFromString(Configuration::get<std::string>("log.logfilelevel"));
+    filelog_level = level;
+
+    if(filelog_level == LogLevel::OFF)
+    	return;
+
+    if(filelog_stream.is_open()){
+    	throw MustNotHappenException("File logging was already enabled.");
+    }
+
+	auto t = std::time(nullptr);
+	auto tm = *std::localtime(&t);
+	std::stringstream file_name;
+	file_name << "log_";
+	file_name << std::put_time(&tm, "%d-%m-%Y_%H-%M-%S");
+	file_name << ".txt";
+
+	namespace bf = boost::filesystem;
+	bf::path path(Configuration::get<std::string>("log.logfilelocation"));
+	if(!bf::exists(path)){
+        bf::create_directory(path);
+	}
+	path /= file_name.str();
+	filelog_stream.open(path.string());
```

Comment:
please check if the stream creation was successful using `filelog_stream.fail()`